src/lib/libast/regex/regcomp.c

```diff
@@ -1294,7 +1294,7 @@ static_fn Rex_t *regcomp_bra(Cenv_t *env) {
 
     if (complicated) {
         Dt_t *dt;
-        Cchr_t *cc;
+        Cchr_t *cc, *cc_begin = 0;
```

Comment:
I instrumented the code and it appears you dealt with all the leaks of this particular pointer. However, I'd like to propose a slightly different approach. The only reason you needed to introduce `cc_begin` is because `cc` is used for two (or more) different purposes. Which is a dangerous anti-pattern.  Don't introduce an alias. Instead define a new var at the top of the function that is initialized to NULL; perhaps named `cc_keys` rather than `cc_begin`. Then assign to that var and use it in the

```
    if (ast.locale.serial != lc_collate_serial || !dt) {
    }
```

block of code. Then simply call `free(cc_keys);` before each `return` in this function.

Also, notice that it appears the `if (dt) {` that immediately follows the preceding block is always true. Both based on analysis of the code and empirical testing by adding a `assert(dt);` just before the `if (dt) {`.

Reply:
> The only reason you needed to introduce cc_begin is because cc is used for two (or more) different purposes. Which is a dangerous anti-pattern. Don't introduce an alias. Instead define a new var at the top of the function that is initialized to NULL; perhaps named cc_keys rather than cc_begin.

How's this approach different from my current approach ? I am freeing this variable at each return from this code block.

Reply:
My proposal simplifies the code on a conceptual level by eliminating the aliasing of the `cc` variable to mean different things. Also, the addition of a pointer on the stack does not matter from a performance standpoint in this specific use case. My approach means you only need to call `free(cc_keys);` before each `return ...;` rather than adding a lot of `free()` calls before `goto...;` statements. Etcetera.

Your solution fixes the problem. It also further enshrines an anti-pattern. When fixing a bug in the code we should also eliminate bad practices (i.e., anti patterns) when possible.